master/buildbot/www/rest.py

```diff
@@ -249,6 +249,15 @@ def checkFields(fields, negOk=False):
                 except Exception:
                     raise BadRequest('invalid offset')
                 continue
+            # As discussed with @tardyp, this is the best way to implement
+            # properties behavior.
+            elif arg in ('property',):
+                try:
+                    properties = [v.decode('utf-8') for v in reqArgs[arg]]
+                except Exception:
+                    raise BadRequest('invalid property value for %s' % arg)
+                filters.append(resultspec.Filter(arg, 'eq', properties))
```

Comment:
@tardyp
I may not get something, but from my understanding, **fields** are tight to **entityType**, which is not the case here, there's no **property** field in ``builds`` table!

As far as I know, `there's only a FK in builds_properties table on builds one`. Which allow to get only one build's properties at a time (knowing the build id).

Actually, you suggested me to *join/aggregate* builds and their *filtered* properties at ``Data API`` level, which might not be the best  way to do what you're suggesting...